src/server/pfs/cmds/cmds.go

```diff
@@ -1071,47 +1079,105 @@ $ {{alias}} foo@master:path1 bar@master:path2`,
 					return err
 				}
 			}
-
 			c, err := client.NewOnUserMachine("user")
 			if err != nil {
 				return err
 			}
 			defer c.Close()
 
-			newFiles, oldFiles, err := c.DiffFile(
-				newFile.Commit.Repo.Name, newFile.Commit.ID, newFile.Path,
-				oldFile.Commit.Repo.Name, oldFile.Commit.ID, oldFile.Path,
-				shallow,
-			)
-			if err != nil {
-				return err
-			}
-
-			if len(newFiles) > 0 {
-				fmt.Println("New Files:")
-				writer := tabwriter.NewWriter(os.Stdout, pretty.FileHeader)
-				for _, fileInfo := range newFiles {
-					pretty.PrintFileInfo(writer, fileInfo, fullTimestamps, false)
+			return pager.Page(noPager, os.Stdout, func(w io.Writer) (retErr error) {
```

Comment:
The `nameOnly` and `!nameOnly` paths appear to be fairly distinct here.  It might be cleaner to have a function like:
```go
func forEachFile(oldFiles []*FileInfo, newFiles []*FileInfo, cb func(oldFileInfo *FileInfo, newFileInfo *FileInfo) error) {...}
```

Then branch on `nameOnly` just once at the top-level.

Reply:
Yeah, that sounds like a better idea, I'll do this.